### PR TITLE
feat: Add PromptGuard integration

### PR DESCRIPTION
**Pull request by zizhong**
Add PromptGuard integration
-------
There are two approaches to integrate PromptGuard with a LangChain application.

1. PromptGuardLLMWrapper
2. functions that can be used in LangChain expression.

-----
- Dependencies
`promptguard` python package, which is a runtime requirement if you'd try out the demo.

-  @baskaryan @hwchase17 Thanks for the ideas and suggestions along the development process.

### CODE REVIEW

**Comment by baskaryan** [comment]
made some small updates in #9485, feel free to merge those into here and then i think it looks good!

**Comment by zizhong** [comment]
@baskaryan Thanks for the review and changes! Cherrypicked to this PR.

File: libs/langchain/langchain/callbacks/manager.py
Comment (baskaryan) [inline libs/langchain/langchain/callbacks/manager.py:1]: ah i meant to undo this change in my branch but forgot to — could you undo?

**Review by baskaryan — COMMENTED** [review]
one mistake on my part, otherwise lgtm!

File: libs/langchain/langchain/callbacks/manager.py
Comment (zizhong) [reply libs/langchain/langchain/callbacks/manager.py:1]: @baskaryan Reverted. Nice catch!
